File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int dfs(TreeNode root, int val) {
        if(root == null) {
            return 0;
        }
        else if(root.left == null && root.right == null) {
            return val * 10 + root.val; 
        }
        return dfs(root.left, val * 10 + root.val) + dfs(root.right, val * 10 + root.val);
    }
    public int sumNumbers(TreeNode root) {
        return dfs(root, 0);
    }
}

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getSumNumbers(self, root: Optional[TreeNode], val: int):
        if root is None:
            return
        val = val * 10 + root.val
        if root.left is None and root.right is None:
            self.res = self.res + val
        self.getSumNumbers(root.left, val)
        self.getSumNumbers(root.right, val)

    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        self.res = 0
        self.getSumNumbers(root, 0)
        return self.res

        